x-pack/plugins/alerting/server/application/rule/methods/bulk_delete/bulk_delete.ts

```diff
@@ -8,27 +8,35 @@ import pMap from 'p-map';
 import { KueryNode, nodeBuilder } from '@kbn/es-query';
 import { SavedObjectsBulkUpdateObject } from '@kbn/core/server';
 import { withSpan } from '@kbn/apm-utils';
-import { RawRule } from '../../types';
-import { convertRuleIdsToKueryNode } from '../../lib';
-import { bulkMarkApiKeysForInvalidation } from '../../invalidate_pending_api_keys/bulk_mark_api_keys_for_invalidation';
-import { ruleAuditEvent, RuleAuditAction } from '../common/audit_events';
-import { tryToRemoveTasks } from '../common';
-import { API_KEY_GENERATE_CONCURRENCY } from '../common/constants';
+import { RawRule } from '../../../../types';
```

Comment:
We should move away from the old types `RawRule` in favor of the new types `RuleAttributes`. Recall that we want to make sure we transform the rules from the application layer to the data layer for deletion.

Reply:
Is it what you mean? https://github.com/elastic/kibana/pull/164015/commits/4d3d6baced97e82ab53373df271f453923ae3d9c

Reply:
And also what's the different between RawRule and RuleAttributes. Can I just to write RuleAttributes instead of RawRule? Or I need to rewrite API code?